### ISSUE TITLE
[BUG] NoSuchMethodError: java.nio.file.Path.of(Ljava/lang/String;[Ljava/lang/String;) on Java 1.8

### ISSUE THREAD

**Issue by chenwei182729** [opensearch-project/OpenSearch#1591]
There are something wrong when i run openseach on jvm 1.8

1. jvm

```
java version "1.8.0_301"
Java(TM) SE Runtime Environment (build 1.8.0_301-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.301-b09, mixed mode)
```
2. os

```bash
Linux 5.11.0-40-generic #44~20.04.2-Ubuntu SMP Tue Oct 26 18:07:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
3. openstack version 
`opensearch-1.1.0-linux-x64.tar.gz`
4. error info

```bash
java.lang.NoSuchMethodError: java.nio.file.Path.of(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path;
	at org.opensearch.notebooks.settings.PluginSettings.<clinit>(PluginSettings.kt:269) ~[?:?]
	at org.opensearch.notebooks.NotebooksPlugin.getSettings(NotebooksPlugin.kt:79) ~[?:?]
	at org.opensearch.plugins.PluginsService.lambda$getPluginSettings$0(PluginsService.java:107) ~[opensearch-1.1.0.jar:1.1.0]
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267) ~[?:1.8.0_301]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384) ~[?:1.8.0_301]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_301]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_301]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_301]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_301]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499) ~[?:1.8.0_301]
	at org.opensearch.plugins.PluginsService.getPluginSettings(PluginsService.java:107) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.node.Node.<init>(Node.java:398) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.node.Node.<init>(Node.java:301) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:240) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:240) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:406) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:182) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:173) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:99) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140) ~[opensearch-cli-1.1.0.jar:1.1.0]
	at org.opensearch.cli.Command.main(Command.java:103) ~[opensearch-cli-1.1.0.jar:1.1.0]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:139) ~[opensearch-1.1.0.jar:1.1.0]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:105) ~[opensearch-1.1.0.jar:1.1.0]
fatal error in thread [main], exiting
java.lang.NoSuchMethodError: java.nio.file.Path.of(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path;
	at org.opensearch.notebooks.settings.PluginSettings.<clinit>(PluginSettings.kt:269)
	at org.opensearch.notebooks.NotebooksPlugin.getSettings(NotebooksPlugin.kt:79)
	at org.opensearch.plugins.PluginsService.lambda$getPluginSettings$0(PluginsService.java:107)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at org.opensearch.plugins.PluginsService.getPluginSettings(PluginsService.java:107)
	at org.opensearch.node.Node.<init>(Node.java:398)
	at org.opensearch.node.Node.<init>(Node.java:301)
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:240)
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:240)
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:406)
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:182)
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:173)
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:99)
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140)
	at org.opensearch.cli.Command.main(Command.java:103)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:139)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:105)
```

**Comment by anasalkouz** [comment]
Track if we are missed running tests for plugins in Java 1.8

**Comment by anirudha** [comment]
can you pls add steps to reproduce this issue, thanks

**Comment by joshuali925** [comment]
notebooks is merged into observability, but I'm able to confirm the issue on 1.2 observability. We will look into this, as a workaround could you try running it with `JAVA_HOME= ./bin/opensearch`? This worked for me as it's going to use the JDK bundled in OpenSearch, which is JDK 15

**Comment by anirudha** [comment]
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html#of(java.lang.String,java.lang.String...)

is only available in java 11 onwards which is used by many plugins
https://sourcegraph.com/search?q=context:global+repo:github.com/opensearch-project/+%5CbPath%5C.of%5C%28&patternType=regexp

**Comment by anirudha** [comment]
https://github.com/opensearch-project/opensearch-plugins/issues/64
pending for java support on plugins

**Comment by jcgraybill** [comment]
Looks like affected plugins are reports-scheduler, notebooks, observability - from that list above. (Everything else is test, unreleased, or data-prepper.) Pls create issues and fix in all three for the 1.3 release.

**Comment by saratvemulapalli** [comment]
@joshuali925 / @anirudha I see notebooks plugin is already being built for 1.8. But we are using it with JDK 14.
Ref: https://github.com/opensearch-project/notifications/blob/main/notifications/build.gradle#L78
(I couldn't find a repo for observability)

Integration tests are being run with JDK 14. 
Ref: https://github.com/opensearch-project/notifications/blob/main/.github/workflows/notifications-test-and-build-workflow.yml#L38

Yeah looks like even though plugins are being built for JDK 8/ Java version string 1.8, we are only testing it in JDK 14 or the bundled JDK with OpenSearch. 

As @anirudha said opensearch-project/opensearch-plugins#64 should be completed before we can support them.

**Comment by dblock** [comment]
Let's sort this all out for 1.3, starting with https://github.com/opensearch-project/opensearch-plugins/issues/64 across all plugins.

**Comment by dblock** [comment]
https://stackoverflow.com/questions/60616244/replace-nio-file-pathofstring-for-java-8 Should help

**Comment by dblock** [comment]
OpenSearch 1.3.0 has restored JDK8 support via https://github.com/opensearch-project/opensearch-plugins/issues/64. Please reopen if you're still seeing problems on JDK8.
